### PR TITLE
fix(streaming): stop the SSE stale-frontier clear from firing on benign snapshot overlaps

### PR DESCRIPTION
**Pull request by siddseethepalli**
Addresses Codex review feedback on #38483.

## Problem

#38483 added a client-side stale-frontier guard: when a live SSE event's `seq` trailed its conversation's local-seq frontier by ≥ `SSE_REPLAY_RING_COUNT_LIMIT`, the frontier was treated as a stale seq-generation anchor — cleared, and the event applied.

Codex (P2) flagged that gap **size** is not a sound signal. `recordLocalSeq` is also called from the `/messages` snapshot paths (`use-conversation-history.ts:178`, `use-message-reconciliation.ts:118`), and `seq` is a single **global** counter across all conversations. During a bursty turn or a main-thread stall, a snapshot advances the frontier far past where the queued live backlog sits, so a valid old queued frame trails the frontier by more than the ring — with **no daemon reset**. The old heuristic then cleared the frontier and re-applied that frame (and, because the frontier was re-seeded low, the rest of the queued backlog too), duplicating deltas / rolling back chat control state until reconciliation repaired it.

## Invariant chosen

A frontier is a stale (dead-generation) anchor **only** inside the re-climb window of an **observed** generation reset:

- the client saw a backwards `seq` jump (`eventSeq < stored` — the daemon's counter restarted), recording the abandoned generation's ceiling via `recordAbandonedGeneration(stored)`;
- the live cursor has **not yet** re-climbed to that ceiling (`liveCursor < abandonedCeiling`); and
- the frontier sits at or above the ceiling (`localSeq >= abandonedCeiling`).

Only then is the frontier provably from a dead generation the live stream hasn't reached, so clearing + applying is safe. Outside that window (no reset, or the cursor already re-passed the ceiling) a large under-frontier gap is a legitimate snapshot overlap: the events are contained in the snapshot, so they drop as ordinary replays and the frontier is **held** — the whole backlog stays suppressed rather than re-applying. Gap size is no longer consulted for this decision.

The abandoned-ceiling is a seq-generation / connection fact (like the reconnect cursor), so it lives in `reconnect-cursor.ts` and is cleared on assistant switch (a fresh seq space), monotonic across resets.

## Why the stale-anchor self-heal is preserved

The incident #38483 fixed: a warm client observes the daemon crash-restart (backwards `seq` → generation reset), clears its frontiers, then a `/messages` request that raced the reset re-poisons a frontier with the dead generation's anchor. Because the reset **was** observed, `abandonedGenerationCeiling` is set; the re-poisoned frontier sits at the abandoned ceiling while the live cursor is below it → the guard fires, clears the frontier, and live events resume applying. Covered end-to-end by the rewritten test (it now drives a real generation reset first, matching the incident sequence).

## Trade-off (documented in code)

One exotic poison variant is no longer covered by the client guard: a fresh page load where the SSE first event seeds the cursor **before** a stale `/messages` anchor arrives (so no backwards jump is observed) against an un-updated daemon. #38483's daemon-side fix (counter floored above every persisted anchor on restart) eliminates that variant at the source; absent the fix it degrades to pre-#38483 behavior — self-recovers as the counter climbs / on the next reconcile — not a hard regression. This is the deliberate cost of refusing to false-positive on the common benign overlap, which the two cases are otherwise numerically indistinguishable from.

## Tests

- `sse-event-consumer.test.ts`:
  - faithful re-poison-after-reset **heal** (drives a real generation reset first, then a racing stale anchor);
  - **NEW** benign large-overlap-with-no-reset case: the over-ring event drops as an ordinary replay, the frontier is held, a second backlog frame also drops, and `sse_local_seq_stale_generation` does **not** fire;
  - **NEW** window-boundary case: reset observed but the cursor re-climbed past the ceiling → ordinary replay.
- `reconnect-cursor.test.ts`: `recordAbandonedGeneration` monotonicity + cleared on reset.
- Completed the `reconnect-cursor` mock surface in `cold-anchor` / `sse-service` / `stream-transport` tests (shared-process `mock.module`-leakage hygiene; CI runs each file isolated so this only affects local multi-file runs).

Touched web tests pass in CI-style per-file isolation (`bun scripts/run-tests.ts`). `stream-transport.test.ts` / `use-message-reconciliation.test.tsx` fail only on absent gitignored generated code in this worktree — unrelated to this change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by siddseethepalli** [comment]
**Human attention — moderate confidence (streaming-core heuristic).**

This replaces the gap-size heuristic from #38483 with a generation-aware window check. Please focus review on:

- **The invariant** (`sse-event-consumer.ts`, `frontierFromDeadGeneration`): a frontier is a dead-generation anchor **only** when a generation reset was observed, the live cursor is still below the abandoned ceiling, and the frontier sits at/above it. The claim is that the benign case (bursty turn / main-thread stall advancing the frontier past the queued live backlog) and the poison case are numerically indistinguishable from gap size alone — an observed reset is the only sound discriminator. Sanity-check that reasoning.
- **`liveCursor != null` is load-bearing**: without it, JS coerces `null < abandonedCeiling` to `0 < ceiling` (true) and would mis-fire. 
- **The documented trade-off**: one exotic poison variant (fresh load, SSE first-event seeds the cursor before a stale `/messages` anchor, un-updated daemon → no backwards jump observed) is no longer healed by the client guard. #38483's daemon-side floor eliminates it at the source; absent that it degrades to pre-#38483 self-recovery, not a hard regression. If we'd rather keep belt-and-suspenders coverage of that variant, we'd need daemon-supplied generation identity on `/messages` — happy to follow up if reviewers want it.
- **Home of `abandonedGenerationCeiling`**: placed in `reconnect-cursor.ts` so it clears on assistant switch (new seq space) in lockstep with the cursor.

**Comment by siddseethepalli** [comment]
Addressed in https://github.com/vellum-ai/vellum-assistant/pull/38894

File: clients/web/src/domains/chat/streaming/sse-event-consumer.ts [hunk 1]
```diff
@@ -345,7 +347,33 @@ export function createSseEventConsumer(
...
+          // above it. There, a replay drop would silently swallow every live
+          // event until the counter re-passes the stale anchor, so clear the
+          // frontier and apply the live event as the new frontier instead.
+          const abandonedCeiling = getAbandonedGenerationCeiling();
+          const liveCursor = getReconnectCursor();
+          const frontierFromDeadGeneration =
+            abandonedCeiling != null &&
+            liveCursor != null &&
+            liveCursor < abandonedCeiling &&
+            localSeq >= abandonedCeiling;
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply clients/web/src/domains/chat/streaming/sse-event-consumer.ts:375]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/38894
